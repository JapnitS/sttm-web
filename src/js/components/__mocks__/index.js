export * from './shabad';
export * from './multipleShabads';
export * from './store';
