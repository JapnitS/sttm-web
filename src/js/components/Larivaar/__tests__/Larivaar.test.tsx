import React from 'react';
import { Provider } from 'react-redux'
import { mockStore } from '../../__mocks__';

import { render } from '@testing-library/react';
import { Larivaar } from '..';

const store = mockStore({ darkMode: false });

describe('<Larivaar />', () => {
  it('renders correctly', () => {

  })
  // it('renders children when disabled', () => {
  //   const { container } = render(
  //     <Provider store={store}>
  //       <Larivaar enable={false}>Vahi Guru</Larivaar>
  //     </Provider>
  //   );

  //   expect(container).toMatchSnapshot();
  // });

  // it('renders concatenated children when enabled', () => {
  //   const { container } = render(
  //     <Provider store={store}>
  //       <Larivaar>Vahi Guru</Larivaar>
  //     </Provider>
  //   );

  //   expect(container).toMatchSnapshot();
  // });

  // it('renders odd words as orange when larivaarAssist is enabled', () => {
  //   const container = render(
  //     <Provider store={store}>
  //       <Larivaar larivaarAssist={true}>Vahi Guru</Larivaar>
  //     </Provider>
  //   );

  //   expect(container).toMatchSnapshot();
  // });
});
