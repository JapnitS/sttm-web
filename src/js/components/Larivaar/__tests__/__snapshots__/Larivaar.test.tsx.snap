// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Larivaar /> renders children when disabled 1`] = `
<div>
  <span
    class=""
  >
     Vahi 
  </span>
  <span
    class=""
  >
     Guru 
  </span>
</div>
`;

exports[`<Larivaar /> renders concatenated children when enabled 1`] = `
<div>
  <span
    class=""
  >
    <span>
      V
      <wbr />
    </span>
  </span>
  <span
    class=""
  >
    <span>
      a
      <wbr />
    </span>
  </span>
  <span
    class=""
  >
    <span>
      h
      <wbr />
    </span>
  </span>
  <span
    class="larivaar-word"
  >
    <span>
      Gu
      <wbr />
    </span>
  </span>
  <span
    class="larivaar-word"
  >
    <span>
      ru
      <wbr />
    </span>
  </span>
</div>
`;

exports[`<Larivaar /> renders odd words as orange when larivaarAssist is enabled 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class=""
      >
         Vahi 
      </span>
      <span
        class=""
      >
         Guru 
      </span>
    </div>
    <div>
      <span
        class=""
      >
        <span>
          V
          <wbr />
        </span>
      </span>
      <span
        class=""
      >
        <span>
          a
          <wbr />
        </span>
      </span>
      <span
        class=""
      >
        <span>
          h
          <wbr />
        </span>
      </span>
      <span
        class="larivaar-word"
      >
        <span>
          Gu
          <wbr />
        </span>
      </span>
      <span
        class="larivaar-word"
      >
        <span>
          ru
          <wbr />
        </span>
      </span>
    </div>
    <div>
      <span
        class=""
      >
        <span>
          V
          <wbr />
        </span>
      </span>
      <span
        class=""
      >
        <span>
          a
          <wbr />
        </span>
      </span>
      <span
        class=""
      >
        <span>
          h
          <wbr />
        </span>
      </span>
      <span
        class="larivaar-word larivaar-assist-word"
      >
        <span>
          Gu
          <wbr />
        </span>
      </span>
      <span
        class="larivaar-word larivaar-assist-word"
      >
        <span>
          ru
          <wbr />
        </span>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class=""
    >
      <span>
        V
        <wbr />
      </span>
    </span>
    <span
      class=""
    >
      <span>
        a
        <wbr />
      </span>
    </span>
    <span
      class=""
    >
      <span>
        h
        <wbr />
      </span>
    </span>
    <span
      class="larivaar-word larivaar-assist-word"
    >
      <span>
        Gu
        <wbr />
      </span>
    </span>
    <span
      class="larivaar-word larivaar-assist-word"
    >
      <span>
        ru
        <wbr />
      </span>
    </span>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
